DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,324 @@
+
+
+# Register Map and streaming interfaces
+This FPGA tutorial demonstrates how to specify the kernel control interfaces and kernel argument interfaces. The kernel control interfaces can be implemented as register map interfaces or streaming interfaces. Similarly, the kernel argument interfaces can be implemented as register map interfaces or streaming interfaces as well, independent of which interface that the kernel control is using. The register map interface is referring to an interface registered in the kernel memory map, where the streaming interface is referring to an interface that signals are implemented in simple conduits.
```

Comment:
```suggestion
This FPGA tutorial demonstrates how to specify the kernel control interfaces and kernel argument interfaces. The kernel control interfaces can be implemented as register map interfaces or streaming interfaces. Similarly, the kernel argument interfaces can be implemented as register map interfaces or streaming interfaces as well, independent of which interface that the kernel control is using. The register map interface is referring to an interface registered in the kernel memory map, where the streaming interface is referring to an interface that signals are implemented in simple conduits.
```
```suggestion
This FPGA tutorial demonstrates how to specify the kernel control interface and kernel argument interfaces. The kernel control interface (namely, the `start` and `done` signals) can be implemented in the kernel's control/status register (CSR), or using a ready/valid handshake. Similarly, the kernel arguments can be passed through the CSR, or through dedicated streaming interfaces. The control interface and any argument interfaces are independent, so it's possible to implement the control interface with a ready/valid handshake, and implement the kernel arguments in the CSR.
```

Reply:
What 'valid' signal do the conduit inputs synchronize with? that is, when does the attached component need to guarnatee the data on the input conduits is valid? in HLS, we synchronized with the `start` signal, but I don't think we had a good synchronization when the `start` signal was in the CSR. We should be clear about that here. The following configurations are valid:


| Control Interface | Argument Interface | argument interface synchronization       |
|-------------------|--------------------|------------------------------------------|
| streaming         | streaming          | Synchronize with `start` and `ready_out` |
| streaming         | register_map       | N/A                                      |
| register_map      | streaming          | *No synchronization possible*            |
| register_map      | register_map       | N/A                                      |

Reply:
For register_map control and streaming arguments, the sycl IPA behaviour is that it will be consumed on the next clock cycle of writing to the start register. That is, if CSR_ADDR = START_REG && CSR_WRITE = 1 on clock cycle `n`, the streaming arguments are consumed on clock cycle `n+1`. This is the same behaviour as HLS, but I think it's somewhat weird and we may not want to advertise that to the users. So I agree *no synchronization possible* is probably the best.